datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -687,6 +698,8 @@ impl Stream for GroupedHashAggregateStream {
                         Some(Ok(batch)) => {
                             let timer = elapsed_compute.timer();
 
+                            let batch = self.filter_by_selection_vector(batch)?;
```

Comment:
So instead of this - it should compute the indices and only do work for `group_aggregate_batch` (I think there need to be some more changes there).